entrypoint.sh

```diff
@@ -11,6 +11,11 @@ function appendParams(){
 
 kn_command=("kn" "service")
 
+# get current namespace
+current_namespace=$(oc config view --minify --output 'jsonpath={..namespace}')
+
+INPUT_SERVICE_NAMESPACE=${INPUT_SERVICE_NAMESPACE:-$current_namespace}
```

Comment:
I don't think this is the right behaviour.

The `oc config view` command above may return nothing. It's possible (and common) to not have a namespace set in your current context.

Also, passing `--namespace=<current context's namespace` is the same as not passing `--namespace` at all.

All you have to do is something like:

```
if [[ -n $INPUT_SERVICE_NAMESPACE ]]; then
    echo "Set service namespace to '$INPUT_SERVICE_NAMESPACE'"
    appendParams "--namespace=$INPUT_SERVICE_NAMESPACE"
else
    echo "No namespace provided"
fi

Reply:
also be sure to update the `describe` to use/not use the same namespace.